### PR TITLE
Add description for more workflows

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Some workflows are not documented in `MAINTENANCE.md`. This PR adds short descriptions for them.

I also add indentations to the description of other workflows to make it more readable.

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: MAINTENANCE.md [hunk 1]
```diff
@@ -47,43 +47,58 @@ conda and the `Makefile` to run the tests and checks.
...
 ### GitHub Actions
 
-There are 5 configuration files located in `.github/workflows`:
+There are 8 configuration files located in `.github/workflows`:
 
 1. `style_checks.yaml` (Code lint and style checks)
 
-This is ran on every commit to the *master* and Pull Request branches.
-It is also scheduled to run daily on the *master* branch.
+   This is ran on every commit to the *master* and Pull Request branches.
```
Comment (meghanrjones) [inline MAINTENANCE.md:54]:
```suggestion
   This is run on every commit to the *master* and Pull Request branches.
```

File: MAINTENANCE.md [hunk 2]
```diff
@@ -47,43 +47,58 @@ conda and the `Makefile` to run the tests and checks.
...
+   It is also scheduled to run daily on the *master* branch.
 
 2. `ci_tests.yaml` (Tests on Linux/macOS/Windows)
 
-This is ran on every commit to the *master* and Pull Request branches.
-It is also scheduled to run daily on the *master* branch.
-In draft Pull Requests, only one job (Ubuntu + Python latest)
-is triggered to save on Continuous Integration resources.
+   This is ran on every commit to the *master* and Pull Request branches.
```
Comment (meghanrjones) [inline MAINTENANCE.md:59]:
```suggestion
   This is run on every commit to the *master* and Pull Request branches.
```

File: MAINTENANCE.md [hunk 3]
```diff
@@ -47,43 +47,58 @@ conda and the `Makefile` to run the tests and checks.
...
-On the *master* branch, the workflow also handles the documentation deployment:
+   On the *master* branch, the workflow also handles the documentation
+   deployment:
 
-* Updating the development documentation by pushing the built HTML pages from the
-  *master* branch onto the `dev` folder of the *gh-pages* branch.
-* Updated the `latest` documentation link to the new release.
+   * Updating the development documentation by pushing the built HTML pages
+     from the *master* branch onto the `dev` folder of the *gh-pages* branch.
+   * Updated the `latest` documentation link to the new release.
```
Comment (meghanrjones) [inline MAINTENANCE.md:69]:
```suggestion
   * Updating the `latest` documentation link to the new release.
```

File: MAINTENANCE.md [hunk 4]
```diff
@@ -47,43 +47,58 @@ conda and the `Makefile` to run the tests and checks.
...
+   This is triggered when a PR is marked as "ready for review", or using the
+   slash command `/test-gmt-dev`. It is also scheduled to run daily on the
+   *master* branch.
 
 4. `cache_data.yaml` (Caches GMT remote data files needed for GitHub Actions CI)
 
-This is scheduled to run every Sunday at 12 noon.
-If new remote files are needed urgently, maintainers can manually uncomment
-the 'pull_request:' line in that `cache_data.yaml` file to refresh the cache.
+   This is scheduled to run every Sunday at 12 noon.
```
Comment (meghanrjones) [inline MAINTENANCE.md:79]: UTC time zone?

File: MAINTENANCE.md [hunk 5]
```diff
@@ -47,43 +47,58 @@ conda and the `Makefile` to run the tests and checks.
...
+   If new remote files are needed urgently, maintainers can manually uncomment
+   the 'pull_request:' line in that `cache_data.yaml` file to refresh the cache.
 
 5. `publish-to-pypi.yml` (Publish wheels to PyPI and TestPyPI)
 
-This workflow is ran to publish wheels to PyPI and TestPyPI (for testing only).
-Archives will be pushed to TestPyPI on every commit to the *master* branch and
-tagged releases, and to PyPI for tagged releases only.
+   This workflow is ran to publish wheels to PyPI and TestPyPI (for testing only).
```
Comment (meghanrjones) [inline MAINTENANCE.md:85]:
```suggestion
   This workflow is run to publish wheels to PyPI and TestPyPI (for testing only).
```

File: MAINTENANCE.md [hunk 6]
```diff
@@ -47,43 +47,58 @@ conda and the `Makefile` to run the tests and checks.
...
+   Archives will be pushed to TestPyPI on every commit to the *master* branch
+   and tagged releases, and to PyPI for tagged releases only.
 
+6. `release-drafer.yml` (Drafts the next release notes)
+
+    This workflow is ran to update the next releases notes as pull requests are
```
Comment (meghanrjones) [inline MAINTENANCE.md:91]:
```suggestion
    This workflow is run to update the next releases notes as pull requests are
```

File: MAINTENANCE.md [hunk 7]
```diff
@@ -47,43 +47,58 @@ conda and the `Makefile` to run the tests and checks.
...
+    merged into master.
+
+7. `check-links.yml` (Check links in the repository and website)
+
+   This workflow is ran weekly to check all external links in plaintext and
```
Comment (meghanrjones) [inline MAINTENANCE.md:96]:
```suggestion
   This workflow is run weekly to check all external links in plaintext and
```

File: MAINTENANCE.md [hunk 8]
```diff
@@ -47,43 +47,58 @@ conda and the `Makefile` to run the tests and checks.
...
+   This is triggered when a PR is marked as "ready for review", or using the
+   slash command `/test-gmt-dev`. It is also scheduled to run daily on the
+   *master* branch.
 
 4. `cache_data.yaml` (Caches GMT remote data files needed for GitHub Actions CI)
 
-This is scheduled to run every Sunday at 12 noon.
-If new remote files are needed urgently, maintainers can manually uncomment
-the 'pull_request:' line in that `cache_data.yaml` file to refresh the cache.
+   This is scheduled to run every Sunday at 12 noon.
```
Comment (seisman) [inline MAINTENANCE.md:79]:
```suggestion
   This is scheduled to run every Sunday at 12:00 (UTC).
```

File: MAINTENANCE.md (hunk 4 above)
Comment (seisman) [reply MAINTENANCE.md:79]: Yes, I changed it to "This is scheduled to run every Sunday at 12:00 (UTC)."

**Review by meghanrjones — APPROVED** [review]

**Pull request merged**